cmd/samToWig/samToWig.go

```diff
@@ -45,15 +45,16 @@ func samToWig(samFileName string, reference string, outfile string, paired bool,
 	} else {
 		for aln, done = sam.NextAlignment(samFile); done != true; aln, done = sam.NextAlignment(samFile) {
 			currentBed = convert.SamToBed(aln)
+			//} this bracket shouldn't be here! checking currentBed != nil should happen within for loop. Otherwise check will happen AFTER the entire sam file was read, and only the last read will be appended to outBed, leading to mostly 0 in the bed and wig files
+			if currentBed != nil {
+				outBed = append(outBed, currentBed)
+			}
 		}
-		if currentBed != nil {
-			outBed = append(outBed, currentBed)
-		}
-	}
 
-	outWig = convert.BedReadsToWig(outBed, ref)
-	log.Printf("Length of outWig: %d", len(outWig))
-	wig.Write(outfile, outWig)
+		outWig = convert.BedReadsToWig(outBed, ref)
+		log.Printf("Length of outWig: %d", len(outWig))
+		wig.Write(outfile, outWig)
+	}
```

Comment:
I think this bracket should be moved 3 lines up so that the 3 lines above this line is not in the "else".  This way those three lines will run for both when the fragment length is given, or when that option is not used.